adapters/openrtb_util.go

```diff
@@ -6,32 +6,107 @@ import (
 	"github.com/prebid/openrtb"
 )
 
-func makeOpenRTBGeneric(req *pbs.PBSRequest, bidder *pbs.PBSBidder, bidderFamily string) openrtb.BidRequest {
+func mediaTypeInSlice(t pbs.MediaType, list []pbs.MediaType) bool {
+	for _, b := range list {
+		if b == t {
+			return true
+		}
+	}
+	return false
+}
+
+func makeBanner(unit pbs.PBSAdUnit) *openrtb.Banner {
+	return &openrtb.Banner{
+		W:        unit.Sizes[0].W,
+		H:        unit.Sizes[0].H,
+		Format:   unit.Sizes,
+		TopFrame: unit.TopFrame,
+	}
+}
+
+func makeVideo(unit pbs.PBSAdUnit) *openrtb.Video {
+	mimes := make([]string, len(unit.Video.Mimes))
+	copy(mimes, unit.Video.Mimes)
+	pbm := make([]int8, 1)
+	pbm[0] = unit.Video.PlaybackMethod
+	return &openrtb.Video{
+		MIMEs:          mimes,
+		MinDuration:    unit.Video.Minduration,
+		MaxDuration:    unit.Video.Maxduration,
+		W:              unit.Sizes[0].W,
+		H:              unit.Sizes[0].H,
+		StartDelay:     unit.Video.Startdelay,
+		PlaybackMethod: pbm,
+	}
+}
 
-	imps := make([]openrtb.Imp, len(bidder.AdUnits))
-	for i, unit := range bidder.AdUnits {
+func makeOpenRTBGeneric(req *pbs.PBSRequest, bidder *pbs.PBSBidder, bidderFamily string, mediatypes []pbs.MediaType, singleMediaTypeImp bool) openrtb.BidRequest {
+
+	imps := make([]openrtb.Imp, len(bidder.AdUnits)*len(mediatypes))
```

Comment:
wouldn't it be simpler to turn `mediaTypeInSlice()` into an intersection function that returns a slice (or set) that you could then iterate over below? That way you wouldn't need the `impsPresent` logic since `imps` will be the right size from the start.